# Cohen's h
*By P. Stikker*<br>
https://PeterStatistics.com<br>
https://www.youtube.com/stikpet

Libraries and modules used in this notebook:

In [1]:
import pandas as pd
import numpy as np
import pyreadstat  # to load the example data
import math # optional, in the final notes needed

Cohen's h is suggested as an effect size measures to be used with dichotomous/binary data, by Rosnow and Rosenthal (2003, p. 225). This effect size can be used when comparing two percentages. It is different from Cohen's g which compares a percentage to 50%. A separate video is available for Cohen's g.

Cohen's h can be calculated by (Cohen, 1988, p. 181):

\begin{equation*}
h = \phi_1 - \phi_2
\end{equation*}

Where:

\begin{equation*}
\phi_i = 2 \text{ arcsin} \sqrt{P_i}
\end{equation*}

With ($P_i$) being the actual sample proportion for category $i$.

Lets take a look at an example. We begin with loading some data:

In [2]:
myDf, meta = pyreadstat.read_sav("StudentStatistics.sav")

The field of interest in this data is the 'Gen_Gender', so lets take a look at that one:

In [3]:
myCd = myDf['Gen_Gender'].value_counts()
myCd

2.0    34
1.0    12
Name: Gen_Gender, dtype: int64

We'll need the sum and pick one of the two, so lets grab those:

In [4]:
mySum = sum(myCd.values)
myX = myCd.values[0]

print(mySum)
print(myX)

46
34


Now lets dive into that formula. First we need the two proportions from the sample ($P_1$ and $P_2$):

In [5]:
myP1 = myX / mySum
myP2 = (mySum - myX) / mySum

Then calculate those $\phi_i$, using $\phi_i = 2 \text{ arcsin} \sqrt{P_i}$. This requires the arcsin function, which is available in the numpy library:

In [6]:
phi1 = 2*np.arcsin(myP1**0.5)
phi2 = 2*np.arcsin(myP2**0.5)

Last step is to calculate Cohen's h using $h = \phi_1 - \phi_2$:

In [7]:
CohenH = phi1 - phi2
CohenH

0.9973466941237803

Cohen actually gives thresholds for 'small', 'medium', and 'large' effect sizes for this h at .20, .50, and .80 resp. (Cohen, 1988, pp. 184-185). I would interpret this as:

|\|Cohen h\|| Interpretation|
|-------|---------------|
|0.00 < .20| Negligible|
|0.20 < 0.50 |Small|
|0.50 < 0.80| Medium|
|0.80 or more| Large|

The 0.997 in the example would be considered 'Large'. We can of course let Python do the 'looking in a table' for us:

In [8]:
CohenHnonDir = abs(CohenH)
if CohenHnonDir < .20:
    print('Negligible')
elif CohenHnonDir < .50:
    print('Small')
elif CohenHnonDir < .80:
    print('Medium')
else:
    print('Large')    

Large


In case you prefer not to do this each time, below a quick function:

In [9]:
# this function takes two proportions (a value between 0 and 1) as input and will return Cohen's nondirectional h, and an interpretation.

def cohenH(prop1, prop2):
    phi1 = 2*np.arcsin(prop1**0.5)
    phi2 = 2*np.arcsin(prop2**0.5)    
    cohensHdir = phi1 - phi2
    cohensHnondir = abs(cohensHdir)
    print('Cohen\'s directional h: = ' + str(cohensHdir))
    print('Cohen\'s nondirectional h: = ' + str(cohensHnondir))

    # for the interpretation
    myText = 'Using Cohen\'s qualification this could be considered: '
    if CohenHnonDir < .05:
        qual = 'Negligible'
    elif CohenHnonDir < .15:
        qual = 'Small'
    elif CohenHnonDir < .25:
        qual = 'Medium'
    else:
        qual = 'Large'
    print(myText + qual)
    
    # return Cohen's h and qualification
    return cohensHdir, qual

Lets see if it works:

In [10]:
cohenH(myP1, myP2)

Cohen's directional h: = 0.9973466941237803
Cohen's nondirectional h: = 0.9973466941237803
Using Cohen's qualification this could be considered: Large


(0.9973466941237803, 'Large')

*Final remarks:*

The arcsin calculation can also be done with the math library:

In [11]:
import math

math.asin(myP1**0.5)

1.0347348369283933

In [12]:
np.arcsin(myP1**0.5)

1.0347348369283933

## References
Cohen, J. (1988). *Statistical power analysis for the behavioral sciences* (2nd ed). Hillsdale, N.J: L. Erlbaum Associates.

Rosnow, R. L., & Rosenthal, R. (2003). Effect sizes for experimenting psychologists. *Canadian Journal of Experimental Psychology/Revue Canadienne de Psychologie Expérimentale, 57*(3), 221–237. doi:10.1037/h0087427